instrumentation/github.com/tidwall/buntdb/splunkbuntdb/config.go

```diff
@@ -0,0 +1,176 @@
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+
+	splunkotel "github.com/signalfx/splunk-otel-go"
+)
+
+// instrumentationName is the instrumentation library identifier for a Tracer.
+const instrumentationName = "github.com/signalfx/splunk-otel-go/instrumentation/github.com/tidwall/buntdb/splunkbuntdb"
+
+type config struct {
+	ctx              context.Context
+	tracer           trace.Tracer
+	defaultStartOpts []trace.SpanStartOption
+}
+
+func newConfig(options ...Option) *config {
+	var c config
+	for _, o := range options {
+		if o != nil {
+			o.apply(&c)
+		}
+	}
+
+	attrs := []attribute.KeyValue{
+		semconv.DBSystemKey.String("buntdb"),
+		//attribute.Key("db.buntdb.file").String(":memory:"), // TODO: is this what we want?
+	}
+
+	c.defaultStartOpts = []trace.SpanStartOption{
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	}
+
+	if c.tracer == nil {
+		c.tracer = otel.Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(splunkotel.Version()),
+		)
+	}
+
+	if c.ctx == nil {
+		c.ctx = context.Background()
+	}
+
+	return &c
+}
+
+// copyConfig is used internally to create new config instance
+func copyConfig(c *config) *config {
+	copyOpts := make([]trace.SpanStartOption, len(c.defaultStartOpts))
+	copy(copyOpts, c.defaultStartOpts)
+
+	newCfg := &config{
+		ctx:              c.ctx,
+		tracer:           c.tracer,
+		defaultStartOpts: copyOpts,
+	}
+	return newCfg
+}
+
+// resolveTracer returns an OTel tracer from the appropriate TracerProvider.
+//
+// If the passed context contains a span, the TracerProvider that created the
+// tracer that created that span will be used. Otherwise, the TracerProvider
+// from c is used.
+func (c *config) resolveTracer(ctx context.Context) trace.Tracer {
+	if span := trace.SpanFromContext(ctx); span.SpanContext().IsValid() {
+		return span.TracerProvider().Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(splunkotel.Version()),
+		)
+	}
+	// There is a possibility that the config was not created with newConfig
+	// (i.e. new(Client)), try to handle this situation gracefully.
+	if c == nil || c.tracer == nil {
+		return otel.Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(splunkotel.Version()),
+		)
+	}
+	return c.tracer
+}
+
+// withSpan wraps the function f with a span.
+func (c *config) withSpan(spanName string, f func() error, opts ...trace.SpanStartOption) error {
+	var o []trace.SpanStartOption
+	if c == nil || len(c.defaultStartOpts) == 0 {
+		o = make([]trace.SpanStartOption, len(opts))
+		copy(o, opts)
+	} else {
+		o = make([]trace.SpanStartOption, len(c.defaultStartOpts)+len(opts))
+		copy(o, c.defaultStartOpts)
+		copy(o[len(c.defaultStartOpts):], opts)
+	}
+
+	extraAttrs := []attribute.KeyValue{
+		semconv.DBOperationKey.String(spanName),
+	}
+
+	traceAttr := trace.WithAttributes(extraAttrs...)
+
+	o = append(o, traceAttr)
```

Comment:
Allocating the known capacity of the options slice from the start will ensure the underlying array is the exact size needed.


```suggestion
		o = make([]trace.SpanStartOption, len(opts), len(opts)+1)
		copy(o, opts)
	} else {
		n := len(c.defaultStartOpts)+len(opts)
		o = make([]trace.SpanStartOption, n, n+1)
		copy(o, c.defaultStartOpts)
		copy(o[len(c.defaultStartOpts):], opts)
	}
	o = append(o, trace.WithAttributes(semconv.DBOperationKey.String(spanName)))
```

It might also make more sense to pass `semconv.DBOperationKey.String(spanName))` as an argument to this method. This will avoid this sizing logic and ensure future use of `withSpan` does not mistakenly have non-DB operation spans annotated as DB operations.